library/src/scalaShadowing/language.scala

```diff
@@ -218,6 +218,24 @@ object language {
   /** Where imported, auto-tupling is disabled */
   object noAutoTupling
 
-  /** Where imported loose equality using eqAny is disabled */
+  /** Where imported, loose equality using eqAny is disabled */
   object strictEquality
+
+  /** Where imported, ad hoc extensions of non-open classes in other
+   *  compilation units are allowed.
+   *
+   *  '''Why control the feature?''' Ad-hoc extensions should usually be avoided
+   *  since they typically cannot reply on an "internal" contract between a class
+   *  and its extensions. Only open classes need to specify such a contract.
+   *  Ad-hoc extensions might break for future versions of the extended class,
+   *  since the extended class is free to change its implementation without
+   *  being constrained by an internal contract.
+   *
+   *  '''Why allow it?''' An ad-hoc extension can sometimes be necessary,
+   *  for instance when mocking a class in a testing framework, or to work
+   *  around a bug or missing feature in the original class. Nevertheless,
+   *  such extensions should be limited in scope and clearly documented.
+   *  That's why the language import is required for them.
+   */
+  object adhocExtensions
```

Comment:
I'm not sure `adhocExtensions"`is the best name to use here, it sounds like something related to extension methods. I suggest something more descriptive like `openAllClasses`